File: ScriptService/Dto/Workflows/IndexTransition.cs
namespace ScriptService.Dto.Workflows {
    /// <summary>
    /// argument used to create a transition
    /// </summary>
    public class IndexTransition : Transition {

        /// <summary>
        /// index of node from which transition originates
        /// </summary>
        public int OriginIndex { get; set; }

        /// <summary>
        /// index of node to which transition leads
        /// </summary>
        public int TargetIndex { get; set; }
    }
}

File: nc-scripts/src/app/dto/workflows/indexTransition.ts
import { TransitionType } from './transitionType';

/**
 * transition using node indices as arguments
 */
export interface IndexTransition {

    /**
     * index of node where transition originates from
     */
    originIndex: number,

    /**
     * index of node which is targeted by transition
     */
    targetIndex: number,

    /**
     * condition for transition to execute
     */
    condition?: string,

    /**
     * determines whether transition is used for error handling
     */
    type: TransitionType,

    /**
     * data to log when using this transition
     */
    log?: string
}